Tranga/Server.cs

```diff
@@ -102,6 +102,22 @@ private void HandleRequest(HttpListenerContext context)
         return ret;
     }
 
+    private Dictionary<string, string> GetRequestBody(HttpListenerRequest request)
+    {
+        if (!request.HasEntityBody)
+        {
+            Log("No request body");
+            Dictionary<string, string> emptyBody = new();
+            return emptyBody;
+        }
+        Stream body = request.InputStream;
+        Encoding encoding = request.ContentEncoding;
+        StreamReader reader = new StreamReader(body, encoding);
+        string s = reader.ReadToEnd();
+        Dictionary<string, string> requestBody = JsonConvert.DeserializeObject<Dictionary<string, string>>(s);
+        return requestBody;
+    }
+
     private void HandleGet(HttpListenerRequest request, HttpListenerResponse response)
     {
         Dictionary<string, string> requestVariables = GetRequestVariables(request.Url!.Query);
```

Comment:
Would love to have 
```csharp
Dictionary<string, string> requestVariables = GetRequestVariables(request.Url!.Query);  //Variables in the URI
Dictionary<string, string> requestBody = GetRequestBody(request);                       //Variables in the JSON body
Dictionary<string, string> requestParams = new();                                       //The actual variable used for the API

//Concatenate the two dictionaries for compatibility with older versions of front-ends
requestParams = requestVariables.Concat(requestBody).ToDictionary(x => x.Key, x => x.Value);
```

For GET Requests too

Reply:
GET requests don't support a body

Reply:
[Yesn't](https://developer.mozilla.org/en-US/docs/Web/HTTP/Methods/GET)

> Note: Sending body/payload in a GET request may cause some existing implementations to reject the request — while not prohibited by the specification, the semantics are undefined. It is better to just avoid sending payloads in GET requests.

There's nothing stopping us from doing it, and I tested my implementation without problems.

Reply:
While there's no direct downside to just keeping the parameters in the URI, there has been more than one time where special characters would mess with it.

Reply:
I think the JavaScript fetch method was throwing an error but I'll try it

Reply:
Just read up on it https://developer.mozilla.org/en-US/docs/Web/API/Request/body
Yea it will just return null.
Stupid, but whatever.